client/verta/verta/_deployment/endpoint.py

```diff
@@ -136,15 +137,21 @@ def update(self, run, strategy, wait=False, resources=None, autoscaling=None, en
             raise TypeError("strategy must be an object from verta.deployment.strategies")
 
         # Create new build:
-        url = "{}://{}/api/v1/deployment/workspace/{}/builds".format(
-            self._conn.scheme,
-            self._conn.socket,
-            self.workspace,
-        )
-        response = _utils.make_request("POST", url, self._conn, json={"run_id": run.id})
-        _utils.raise_for_http_error(response)
-        build_id = response.json()["id"]
+        build_id = self._create_build(run_id=run.id)
+        return self._update_from_build(build_id, strategy, wait, resources, autoscaling, env_vars)
+
+    def update_from_model_version(self, model_version, strategy, wait=False, resources=None, autoscaling=None, env_vars=None):
```

Comment:
Let's re-use the same `update()` method to support both run and model_version. I'd be okay with changing the name of the first parameter to `run_or_model_version` for now. I'll ask around for further opinions.

Reply:
Yep I really wanted to use the same method too =)) We could either rename the first parameter, or have both of them, and check them inside the update method.

Reply:
Renae the first parameter. Did you run into an issue trying to do that?

Reply:
Not really. Just feel a bit iffy about a parameter having potentially two types... Anyway, fixed!

Reply:
The way I see it is that both `ExperimentRun` and `RegisteredModelVersion` are actually subtypes of a "buildable/deployable" type. Conceptually, they both have the same artifacts, and they both represent something that can be deployed in the same way. If Python had traits/typeclasses (or a more robust subtyping system...) and we were rewriting the Client, they'd probably both inherit an interface with `log_model()`, `log_artifact()`, etc.

I think it's within reason for the interface (both theoretically and practically for the user) to take a Run or a ModelVersion for the first arg 😄